pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go

```diff
@@ -45,7 +45,7 @@ var log = logf.Log.WithName("controller_amazoncloudintegration")
 // Add creates a new AmazonCloudIntegration Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager, provider operatorv1.Provider, tsee bool) error {
-	if !tsee {
+	if !tsee || provider != operatorv1.ProviderEKS {
```

Comment:
I don't think this is quite the assertion that I was imagining.

I think this is saying that we still _always_ need to install this CRD on AWS. What I was imagining is that we never require the CRD, but if the CRD exists then we run this controller.

Reply:
I initially looked into implementing it the way you described, but it was a slightly more involved change because you need to update the function call on every controller's Add method. Which I can totally do, but...

>I think this is saying that we still always need to install this CRD on AWS

You're exactly correct here, but I thought that was a pretty fair UX / requirement of the user. After all, the original use case that prompted this PR was "i don't want to apply amazon-specific crd's if i'm not on amazon", which both approaches still accomplish.

Also note that what is there in the PR is in line with the other tech debt that we have accumulated in the preview version of this feature, in that we've already overloaded the EKS provider to mean something it is not: indicating that the AWS CNI plugin is in use.